### PR TITLE
Add module elb_security_group and tests

### PR DESCRIPTION
**Pull request by mmoyle**
##### SUMMARY

Add module

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

elb_security_group

### CODE REVIEW

File: plugins/modules/elb_security_group.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+@AWSRetry.exponential_backoff()
+def elb_sg_attach(elbv2_client, alb_arn, sg_ids):
+    '''Attach security groups to elb and return number of sgs attached '''
+
+    response = elbv2_client.set_security_groups(
+        LoadBalancerArn=alb_arn,
+        SecurityGroups=sg_ids
+    )
+
+    return len(response['SecurityGroupIds'])
```
Comment (tremble) [inline plugins/modules/elb_security_group.py:74]: The number of security groups feels like a strange thing to return.  How about

`from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict` (with the rest of the imports)

```suggestion
    return camel_dict_to_snake_dict(response)
```

You can then do `result = elb_sg_attach(` and have more information (including the list of security groups) available

**Review by tremble — CHANGES_REQUESTED** [review]
Thanks for taking the time to submit this new module.

Most of the suggestions below are nit-picks.  However, there are a couple of things

1. For net-new modules we generally prefer that they support check_mode
2. When no change is made changed should return False
3. Please add a changelog fragment: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

File: plugins/modules/elb_security_group.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    elbv2_client = module.client('elbv2')
+
+    try:
+        result['response'] = elb_sg_attach(
+            elbv2_client,
+            module.params.get('alb_arn'),
+            module.params.get('security_group_ids')
+        )
+
+        result['changed'] = True
```
[the remark this answers is not in the document]
Comment (mmoyle) [reply plugins/modules/elb_security_group.py:97]: Hi @tremble ,I think a change will be made unless an exception is thrown or the same set of security group ids is supplied. I am not sure but boto3 may even set the same sgs again. The function does not return whether or not no change was made, and may in itself may not be idempotent, of course i can look if needed. I could first inspect what sgs are already attached and compare against them, but I am not sure if this information is worth the cost of an extra api call. Or? What do you advise in this situation?

**Comment by tremble** [comment]
@mmoyle,

Thanks for taking the time to open this PR.

This PR appears to have stalled out and has a number of open review points.

Looking at the elb_application_lb module we already support managing the Security Groups on an existing ALB.  The existing support also appears to be a little more comprehensive, including supporting using security group names and idempotency.  As such I'm going to close out this PR.

**Pull request closed without merging**